internal/controllers/inventory_controller.go

```diff
@@ -321,50 +323,69 @@ func (t *reconcilerTask) setupDeploymentManagerServerConfig(ctx context.Context,
 	return
 }
 
-// setupAlarmSubscriptionServerConfig creates the resources necessary to start the Alarm Subscription Server.
-func (t *reconcilerTask) setupAlarmSubscriptionServerConfig(ctx context.Context, defaultResult ctrl.Result) (nextReconcile ctrl.Result, err error) {
+// setupAlarmServerConfig creates the resources necessary to start the Alarm Server.
+func (t *reconcilerTask) setupAlarmServerConfig(ctx context.Context, defaultResult ctrl.Result) (nextReconcile ctrl.Result, err error) {
 	nextReconcile = defaultResult
 
-	// Create the needed ServiceAccount.
-	err = t.createServiceAccount(ctx, utils.InventoryAlarmSubscriptionServerName)
+	err = t.createServiceAccount(ctx, utils.InventoryAlarmServerName)
+	if err != nil {
+		t.logger.ErrorContext(
+			ctx,
+			"Failed to deploy ServiceAccount for the alarm server.",
+			slog.String("error", err.Error()),
+		)
+		return
+	}
+
+	err = t.createAlarmServerClusterRole(ctx)
+	if err != nil {
+		t.logger.ErrorContext(
+			ctx,
+			"Failed to create alarm server cluster role",
+			slog.String("error", err.Error()),
+		)
+		return
+	}
+
+	err = t.createAlarmServerClusterRoleBinding(ctx)
```

Comment:
Curious why we need create these for the server? Cant we simply fetch operator's SA and skip creating rbacs and stuff?

Reply:
That's a good question.  They each have their own rbac requirements so from a security point of view of it is best to limit them to only the minimum rules that they need rather than giving all of them the broader permissions of the operator.  Then there would be the issue of the TLS certs/keys.  Our rbac proxy needs to have certs/keys that align with their FQDN hostname so sharing a set of TLS certs/keys across all of the servers wouldn't work properly.  

Unless you had something else in mind I don't see how we can avoid most of the things that are setup on a per-server basis.

Reply:
The operator needs to be granted those permissions if it is to delegate to a server.  i.e., it can't grant something that it itself doesn't already have permission to do.